### PR TITLE
ci: Complete Django 6.1 minimum database coverage

### PR DESCRIPTION
**Pull request by vinitkumar**
Complete the remaining database checks from #8770 after #8774 added PostgreSQL 15. Add one Django 6.1/Python 3.14 job against MariaDB 10.11, verify the connected server's family and minor version, and include its results in combined coverage.

Build checksum-verified SQLite 3.37.2 for the existing Django 6.1/Python 3.12 row. Configure its shared library for the test interpreter and assert the loaded version is exactly 3.37.2 before running the strict-warning suite. Other Django 6.1 SQLite rows assert 3.37+. Retain PostgreSQL 15 and MySQL 8.4 coverage without expanding the Python/database Cartesian matrix.

Validation: workflow YAML and shell syntax pass. The combined 1,811-test suite passes locally on Django 5.2.6, 6.0.8, and 6.1.1. The minimum SQLite check rejects the default 3.50.4 library. A separate full Django 6.1 run uses CPython's SQLite extension rebuilt against SQLite 3.37.2, with deprecation warnings treated as errors. MariaDB execution is covered by the hosted job; no local container runtime is available.

Refs #8770 and #8773. Stacked on #8843 (which includes #8841 and #8842) to keep the database checks and strict Django 6.1 warning gate together.

## Summary by Sourcery

Complete Django 6.1 minimum database compatibility coverage across MariaDB and SQLite in CI.

New Features:
- Add Django 6.1/Python 3.14 coverage against MariaDB 10.11 with server-version validation.

Enhancements:
- Enforce Django 6.1 SQLite minimum-version coverage, including an exact SQLite 3.37.2 check for the Django 6.1/Python 3.12 job.
- Include MariaDB coverage in the combined coverage job while retaining the existing PostgreSQL and MySQL coverage.

Build:
- Build and configure checksum-verified SQLite 3.37.2 for the minimum-version test job.

CI:
- Expand the CI database matrix to validate MariaDB 10.11 and minimum SQLite compatibility without broadening the full Python/database matrix.

Tests:
- Run the Django 6.1 test suite with minimum MariaDB and SQLite versions.

### CODE REVIEW

**Comment by vinitkumar** [comment]
@codex review

**Comment by vinitkumar** [comment]
@codex security review

**Comment by vinitkumar** [comment]
@codex review

**Comment by vinitkumar** [comment]
@codex security review

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -193,6 +238,16 @@ jobs:
...
         uv pip install --system -e .
 
+    - name: Verify Django 6.1 SQLite minimum
+      if: matrix.requirements-file == 'django-6.1.txt'
+      run: |
+        python - <<'PY'
+        import sqlite3
+
+        print(f"SQLite {sqlite3.sqlite_version}")
+        assert sqlite3.sqlite_version_info >= (3, 37), "Django 6.1 requires SQLite 3.37 or newer"
```
[the remark this answers is not in the document]
Comment (vinitkumar) [reply .github/workflows/test.yml:263]: Added a checksum-verified SQLite 3.37.2 build to the existing Django 6.1/Python 3.12 row. LD_LIBRARY_PATH selects that library for the following checks and suite, and the runtime assertion requires exactly 3.37.2. The assertion rejects the default 3.50.4 library. Locally rebuilt the standard CPython SQLite extension against 3.37.2 and ran all 1,811 tests successfully (two expected skips), with upcoming Django deprecation warnings treated as errors.

**Comment by vinitkumar** [comment]
@codex review

**Comment by vinitkumar** [comment]
@codex security review

**Comment by vinitkumar** [comment]
@sourcery-ai review

**Comment by vinitkumar** [comment]
This will make CI more complicated and since Django could already test for this db support, no need of django-cms to duplicate the effort for little gain.
